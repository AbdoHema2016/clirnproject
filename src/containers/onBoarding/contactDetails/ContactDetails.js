import React from 'react';
import {
  View,
  Text,
  KeyboardAvoidingView,
  Platform,
  TouchableOpacity,
  ScrollView,
} from 'react-native';
import Style from './style';
import {connect} from 'react-redux';
import {
  addContactDetailsAction,
  editContactDetailsAction,
  missingContactFieldsAction,
  privacyPolicyCheck,
  getCountryListAction,
} from './redux/actions';
import CLabel from '../../../components/cLabel';
import CChecBox from '../../../components/cCheckBox';
import CPhoneNum from '../../../components/cPhoneNumber';
import CText from '../../../components/cTextField';
import CButton from '../../../components/cButton';
import CPassword from '../../../components/cPassword';
import navigationService from '../../../Navigation/NavigationService';
import CNavigationBackButton from '../../../components/cNavigationBackButton';
import {Constants, testIds} from '../../../utilities';
import {translate} from '../../../Localization';

import HelperFunctions from './HelperFunctions';
const {LOCAL_PATH, defaultCountryCode, defaultCallingCode} = Constants;

class ContactDetails extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      email: {value: '', empty: true, state: ''},
      phone: {value: '', empty: true, state: ''},
      countryCode: {value: defaultCountryCode, empty: false, state: ''},
      callingCode: defaultCallingCode,
      password: {value: '', empty: true, state: '', hidden: true},
    };
    HelperFunctions.setInstance = this;
    HelperFunctions.setProps = this.props;
    this.previousActiveField = null;
    navigationService.navigation = this.props.navigation;
  }

  componentDidMount() {
    this.props.getCodeList();
    this.navigationOptions();
  }

  navigationOptions = () => {
    this.props.navigation.setOptions({
      title: '',
      headerTransparent: true,
      headerLeft: () => (
        <CNavigationBackButton
          title={translate('BACK.BACK_BUTTON_TITLE')}
          backButtonAction={HelperFunctions.goBack}
        />
      ),
    });
  };

  componentDidUpdate(prevProps) {
    if (prevProps.contactDetails !== this.props.contactDetails) {
      HelperFunctions.setProps = this.props;
    }
  }
  onCountryCodeSelect = ({countryCode, callingCode}) => {
    this.setState({callingCode});
    HelperFunctions.editDetails('countryCode', countryCode);
  };
  focusPassword = () => this.password.focus();

  render() {
    const {
      contactDetails: {
        missingFields,
        error,
        policyChecked,
        loading,
        countryCodeList,
      },
    } = this.props;

    const {email, phone, countryCode, password} = this.state;
    return (
      <View style={Style.container}>
        <KeyboardAvoidingView
          style={Style.keyboardAvoidingView}
          behavior={Platform.OS === 'ios' ? 'padding' : null}
          keyboardVerticalOffset={Style.keyboardVerticalOffset.top}>
          <ScrollView
            keyboardDismissMode={true}
            bounces={false}
            alwaysBounceVertical={false}
            showsVerticalScrollIndicator={false}
            keyboardShouldPersistTaps={'handled'}>
            <View>
              <View style={Style.innerContainer}>
                <CLabel
                  style={Style.infoStageLabel}
                  text={translate(
                    'CONTACT_DETAILS_SCREEN_STRINGS.SCREEN_NUMBER',
                  )}
                />
                <CLabel
                  style={Style.appTitle}
                  text={translate(
                    'CONTACT_DETAILS_SCREEN_STRINGS.CONTACT_DETAILS_TITLE',
                  )}
                />
                <View style={Style.inputContainer}>
                  <CLabel
                    style={Style.inputLabel}
                    text={
                      translate('CONTACT_DETAILS_SCREEN_STRINGS.EMAIL_LABEL') +
                      '*'
                    }
                  />
                  <CText
                    testID={testIds.addEmail}
                    accessibilityLabel={testIds.addEmail}
                    reference={(input) => {
                      this.email = input;
                    }}
                    blurOnSubmit={false}
                    onBlur={() =>
                      HelperFunctions.setPreviousActiveFieldStatus(this.email)
                    }
                    onSubmitEditing={HelperFunctions.focusPhoneNumber}
                    returnKeyType={'next'}
                    state={email.state}
                    placeHolderText={translate(
                      'CONTACT_DETAILS_SCREEN_STRINGS.EMAIL_LABEL',
                    )}
                    value={email.value}
                    error={
                      (email.empty && missingFields) ||
                      (error !== '' && error?.email)
                    }
                    keyboardType={'email-address'}
                    onChangeText={(val) =>
                      HelperFunctions.editDetails('email', val, '')
                    }
                    errorMessage={
                      error !== '' && error?.email
                        ? error?.email
                        : translate('LOGIN_SCREEN_STRINGS.ERROR_INVALID_ID')
                    }
                    autoCapitalize={'none'}
                  />

                  <CLabel
                    style={Style.inputLabel}
                    text={
                      translate(
                        'CONTACT_DETAILS_SCREEN_STRINGS.PHONE_NUMBER_LABEL',
                      ) + '*'
                    }
                  />
                  <CPhoneNum
                    testID={testIds.phone}
                    accessibilityLabel={testIds.phone}
                    code={countryCode.value}
                    countryCodeList={countryCodeList}
                    onCountryCodeSelect={this.onCountryCodeSelect}
                    editable={true}
                    onSubmitEditing={this.focusPassword}
                    reference={(input) => {
                      this.phonenumber = input;
                    }}
                    returnKeyType={'next'}
                    state={phone.state}
                    onBlur={() =>
                      HelperFunctions.setPreviousActiveFieldStatus(
                        this.phonenumber,
                      )
                    }
                    placeHolderText={'_ _ _ _ _ _ _ _ _ _'}
                    value={phone.value}
                    maxLength={11}
                    error={
                      (phone.empty && missingFields) ||
                      (error !== '' && error?.mobile)
                    }
                    onChangeText={(val) =>
                      HelperFunctions.editDetails('mobile', val, '')
                    }
                    errorMessage={
                      error !== '' && error?.mobile
                        ? error?.mobile
                        : translate(
                            'CONTACT_DETAILS_SCREEN_STRINGS.EMPTY_PHONE_NUMBER',
                          )
                    }
                    keyboardType={'phone-pad'}
                    accessoryViewLabel={translate('walkthrough.NEXT')}
                    blurOnSubmit={false}
                  />

                  <CLabel
                    style={Style.inputLabel}
                    text={
                      translate('LOGIN_SCREEN_STRINGS.PASSWORD_LABEL') + '*'
                    }
                  />
                  <CPassword
                    testID={testIds.addPassword}
                    accessibilityLabel={testIds.addPassword}
                    blurOnSubmit={true}
                    reference={(input) => {
                      this.password = input;
                    }}
                    onBlur={() =>
                      HelperFunctions.setPreviousActiveFieldStatus(
                        this.password,
                      )
                    }
                    onSubmitEditing={HelperFunctions.submitPasswordEditing}
                    editable={true}
                    value={password.value}
                    hidePasswordAction={HelperFunctions.hidePassword}
                    hidePassword={password.hidden}
                    placeHolderText={translate(
                      'LOGIN_SCREEN_STRINGS.PASSWORD_LABEL',
                    )}
                    state={password.state}
                    customSuperContainer={Style.inputTextsupreContainerStyle}
                    containerStyle={Style.inputTextStyle}
                    returnKeyType={'done'}
                    onChangeText={(val) =>
                      HelperFunctions.editDetails('password', val, '')
                    }
                    error={password.empty && missingFields}
                    errorMessage={translate(
                      'LOGIN_SCREEN_STRINGS.ERROR_INVALID_PASSWORD',
                    )}
                  />

                  <View style={Style.privacyPolicyContainer}>
                    <CChecBox
                      testID={testIds.checkBox}
                      accessibilityLabel={testIds.checkBox}
                      state={policyChecked}
                      textStyle={Style.whiteText}
                      buttonContainerStyle={Style.blackButton}
                      onPress={HelperFunctions.privacyPolicyAction}
                      imagePath={
                        policyChecked
                          ? LOCAL_PATH.PRIVACY_POLICY_CHECKED_ICON
                          : LOCAL_PATH.BLANK_CHECK_BOX_ICON
                      }
                    />
                    <Text
                      style={
                        policyChecked
                          ? Style.IAgreeLabel
                          : Style.IAgreeUncheckedLabel
                      }>
                      {translate('CONTACT_DETAILS_SCREEN_STRINGS.AGREE_LABEL')}{' '}
                    </Text>
                    <TouchableOpacity
                      onPress={HelperFunctions.navToPrivacy}
                      style={Style.privacyPolicyBtn}>
                      <Text
                        style={[
                          policyChecked
                            ? Style.privacyPolicyChecked
                            : Style.privacyPolicyUnchecked,
                          Style.privacyPolicyTxt,
                        ]}>
                        {translate(
                          'CONTACT_DETAILS_SCREEN_STRINGS.PRIVACY_POLICY_LABEL',
                        )}{' '}
                      </Text>
                    </TouchableOpacity>
                  </View>
                  <TouchableOpacity
                    onPress={HelperFunctions.navToTerms}
                    style={Style.termsAndConditionsContainer}>
                    <Text>& </Text>
                    <Text
                      style={[
                        policyChecked
                          ? Style.privacyPolicyChecked
                          : Style.privacyPolicyUnchecked,
                        Style.TermsCondsTxt,
                      ]}>
                      {translate(
                        'CONTACT_DETAILS_SCREEN_STRINGS.T_AND_C_LABEL',
                      )}
                    </Text>
                  </TouchableOpacity>
                </View>
              </View>
            </View>
          </ScrollView>
        </KeyboardAvoidingView>
        <View style={Style.buttonsContainer}>
          <CButton
            testID={testIds.continue}
            accessibilityLabel={testIds.continue}
            text={translate('LOGIN_SCREEN_STRINGS.CONTINUE_LABEL')}
            textStyle={Style.whiteText}
            buttonContainerStyle={Style.blackButton}
            onPress={HelperFunctions.addDetails}
            loading={loading}
            disabled={loading}
            backgroundColor={Style.blackButton.backgroundColor}
          />
        </View>
      </View>
    );
  }
}

const mapStateToProps = (state) => {
  const {
    personalDetails: {access_token, id},
  } = state;
  return {
    userId: id,
    access_token,
    contactDetails: state.contactDetails,
  };
};

const mapDispatchToProps = {
  editContactDetails: editContactDetailsAction,
  addContactetails: addContactDetailsAction,
  missingContactFields: missingContactFieldsAction,
  policyCheck: privacyPolicyCheck,
  getCodeList: getCountryListAction,
};

export default connect(mapStateToProps, mapDispatchToProps)(ContactDetails);
