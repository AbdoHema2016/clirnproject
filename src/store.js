import {createStore, applyMiddleware} from 'redux';
import createSagaMiddleware from 'redux-saga';
import {createLogger} from 'redux-logger';
import rootReducer from './reducers';
import {rootSaga} from './sagas';
const sagaMiddleware = createSagaMiddleware();
const isDevelopmentEnv =
  process && process.env && process.env.NODE_ENV === 'development';
const isProductionEnv =
  process && process.env && process.env.NODE_ENV === 'production';
const logStoreInfo = false;
const middlewares = [sagaMiddleware];
function actionLogger({getState}) {
  return (next) => (action) => {
    let allowed = [
      'jumio',
      'notificationsReducer',
      'otp',
      'companies',
      'signIn',
      'updateContactDetailsReducer',
      'userProfile',
      'vaccinations',
    ];
    let state = getState();
    let loggedStates = Object.keys(state)
      .filter((key) => allowed.includes(key))
      .reduce((obj, key) => {
        return {
          ...obj,
          [key]: state[key],
        };
      }, {});
    console.info({actionType: action.type, loggedStates});
    const returnValue = next(action);
    return returnValue;
  };
}
if (isDevelopmentEnv && logStoreInfo) {
  middlewares.push(createLogger());
}
if (isProductionEnv) {
  middlewares.push(actionLogger);
}

const store = createStore(rootReducer, applyMiddleware(...middlewares));
sagaMiddleware.run(rootSaga);
export default store;
