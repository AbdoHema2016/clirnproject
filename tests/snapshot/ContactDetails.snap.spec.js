import 'react-native';
import React from 'react';
import renderer from 'react-test-renderer';
import {Provider} from 'react-redux';
import store from '../../src/store';
import ContactDetails from '../../src/containers/onBoarding/contactDetails/ContactDetails';

jest.mock('@react-navigation/native', () => {});
jest.mock('react-native-gesture-handler', () => {});
jest.mock('react-native-otp-verify', () => ({
  getHash: jest.fn().mockImplementation(() => Promise.reject()),
  getOtp: jest.fn(),
  removeListener: jest.fn(),
}));

jest.mock('@react-navigation/native', () => ({
  withNavigationFocus: (component) => component,
}));

jest.useFakeTimers();

describe('Contact Details screen', () => {
  const props = {
    navigation: {
      setOptions: jest.fn(),
    },
    editContactDetails: jest.fn(),
    addContactetails: jest.fn(),
    missingContactFields: jest.fn(),
    policyCheck: jest.fn(),
    getCodeList: jest.fn(),
  };
  it('Snapshot ContactDetails', () => {
    const tree = renderer
      .create(
        <Provider store={store}>
          <ContactDetails {...props} />
        </Provider>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
