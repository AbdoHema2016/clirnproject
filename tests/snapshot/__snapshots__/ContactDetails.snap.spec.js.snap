// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Contact Details screen Snapshot ContactDetails 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#F7F7F7",
      "flex": 1,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <RCTScrollView
      alwaysBounceVertical={false}
      bounces={false}
      keyboardDismissMode={true}
      keyboardShouldPersistTaps="handled"
      showsVerticalScrollIndicator={false}
    >
      <View>
        <View>
          <View
            style={
              Object {
                "justifyContent": "space-around",
                "width": 724,
              }
            }
          >
            <Text
              style={
                Array [
                  Object {
                    "color": "#8D8C91",
                    "fontFamily": "SofiaPro-Medium",
                    "fontSize": 16,
                    "marginTop": 160.07999999999998,
                  },
                ]
              }
            >
              [missing "en.CONTACT_DETAILS_SCREEN_STRINGS.SCREEN_NUMBER" translation]
            </Text>
            <Text
              style={
                Array [
                  Object {
                    "color": "#2B342C",
                    "fontFamily": "SofiaPro-Medium",
                    "fontSize": 24,
                    "fontWeight": "600",
                  },
                ]
              }
            >
              [missing "en.CONTACT_DETAILS_SCREEN_STRINGS.CONTACT_DETAILS_TITLE" translation]
            </Text>
            <View
              style={
                Object {
                  "height": "auto",
                  "marginTop": 24,
                  "width": 726,
                }
              }
            >
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2B342C",
                      "fontFamily": "SofiaProRegular",
                      "fontSize": 12,
                      "height": 16,
                      "marginTop": 16,
                    },
                  ]
                }
              >
                [missing "en.CONTACT_DETAILS_SCREEN_STRINGS.EMAIL_LABEL" translation]*
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "height": "auto",
                      "width": 724,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 4,
                        "height": 48,
                        "justifyContent": "center",
                        "marginTop": 4,
                        "width": "100%",
                      },
                      undefined,
                      null,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="addEmail"
                    accessible={true}
                    allowFontScaling={true}
                    autoCapitalize="none"
                    blurOnSubmit={false}
                    keyboardType="email-address"
                    numberOfLines={1}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="[missing \\"en.CONTACT_DETAILS_SCREEN_STRINGS.EMAIL_LABEL\\" translation]"
                    placeholderTextColor="#8D8C91"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    style={
                      Array [
                        Object {
                          "color": "#2B342C",
                          "fontSize": 18,
                          "fontWeight": "500",
                          "lineHeight": 20,
                          "paddingLeft": 19,
                          "paddingRight": 19,
                          "width": "100%",
                        },
                        Object {},
                      ]
                    }
                    testID="addEmail"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2B342C",
                      "fontFamily": "SofiaProRegular",
                      "fontSize": 12,
                      "height": 16,
                      "marginTop": 16,
                    },
                  ]
                }
              >
                [missing "en.CONTACT_DETAILS_SCREEN_STRINGS.PHONE_NUMBER_LABEL" translation]*
              </Text>
              <View
                style={
                  Object {
                    "height": "auto",
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 4,
                        "flexDirection": "row",
                        "height": 48,
                        "marginTop": 4,
                        "paddingHorizontal": 15,
                        "width": "99%",
                      },
                      undefined,
                      undefined,
                      null,
                    ]
                  }
                >
                  <View
                    style={
                      Object {
                        "alignItems": "center",
                        "borderRightColor": "#8D8C91",
                        "borderRightWidth": 1,
                        "flexDirection": "row",
                        "marginRight": 5,
                        "paddingRight": 8,
                      }
                    }
                  >
                    <Image
                      source={
                        Object {
                          "ReactComponent": [Function],
                        }
                      }
                      style={
                        Object {
                          "height": 12,
                          "marginLeft": 5,
                          "marginRight": 3,
                          "width": 12,
                        }
                      }
                    />
                  </View>
                  <TextInput
                    accessibilityLabel="Phone"
                    accessible={true}
                    allowFontScaling={true}
                    blurOnSubmit={false}
                    editable={true}
                    inputAccessoryViewID="phoneInputAccessoryViewID"
                    keyboardType="phone-pad"
                    maxLength={11}
                    numberOfLines={1}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="_ _ _ _ _ _ _ _ _ _"
                    placeholderTextColor="#8D8C91"
                    rejectResponderTermination={true}
                    returnKeyType="next"
                    scrollEnabled={false}
                    style={
                      Array [
                        Object {
                          "bottom": 0,
                          "color": "#2B342C",
                          "flex": 7,
                          "fontFamily": "SofiaProRegular",
                          "fontSize": 18,
                          "height": 48,
                          "paddingLeft": 8,
                          "paddingRight": 19,
                          "paddingTop": 18,
                          "textAlignVertical": "center",
                        },
                      ]
                    }
                    testID="Phone"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                </View>
              </View>
              <Text
                style={
                  Array [
                    Object {
                      "color": "#2B342C",
                      "fontFamily": "SofiaProRegular",
                      "fontSize": 12,
                      "height": 16,
                      "marginTop": 16,
                    },
                  ]
                }
              >
                [missing "en.LOGIN_SCREEN_STRINGS.PASSWORD_LABEL" translation]*
              </Text>
              <View
                style={
                  Array [
                    Object {
                      "height": "auto",
                      "width": 724,
                    },
                    undefined,
                  ]
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#FFFFFF",
                        "borderRadius": 4,
                        "height": 48,
                        "justifyContent": "center",
                        "marginTop": 4,
                        "width": "100%",
                      },
                      undefined,
                      Object {
                        "flexDirection": "row",
                      },
                      null,
                    ]
                  }
                >
                  <TextInput
                    accessibilityLabel="addPassword"
                    allowFontScaling={true}
                    blurOnSubmit={false}
                    editable={true}
                    onBlur={[Function]}
                    onChangeText={[Function]}
                    onSubmitEditing={[Function]}
                    placeholder="[missing \\"en.LOGIN_SCREEN_STRINGS.PASSWORD_LABEL\\" translation]"
                    placeholderTextColor="#8D8C91"
                    rejectResponderTermination={true}
                    returnKeyType="done"
                    secureTextEntry={true}
                    style={
                      Array [
                        Object {
                          "color": "#2B342C",
                          "fontSize": 18,
                          "fontWeight": "500",
                          "lineHeight": 20,
                          "paddingLeft": 19,
                          "paddingRight": 19,
                          "width": "100%",
                        },
                        Object {
                          "marginLeft": 0,
                          "width": "90%",
                        },
                      ]
                    }
                    testID="addPassword"
                    underlineColorAndroid="transparent"
                    value=""
                  />
                  <View
                    accessibilityLabel="Eye"
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "opacity": 1,
                      }
                    }
                    testID="Eye"
                  >
                    <View
                      style={
                        Array [
                          Object {
                            "overflow": "hidden",
                          },
                          Object {
                            "height": 25,
                            "marginRight": 18,
                            "width": 25,
                          },
                        ]
                      }
                    >
                      <FastImageView
                        resizeMode="cover"
                        source={
                          Object {
                            "cache": "immutable",
                            "uri": undefined,
                          }
                        }
                        style={
                          Object {
                            "bottom": 0,
                            "left": 0,
                            "position": "absolute",
                            "right": 0,
                            "top": 0,
                          }
                        }
                      />
                    </View>
                  </View>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "flexDirection": "row",
                    "height": "auto",
                    "marginTop": 16,
                    "width": "100%",
                  }
                }
              >
                <View
                  style={
                    Array [
                      Object {
                        "height": 24,
                        "width": 24,
                      },
                    ]
                  }
                >
                  <View
                    accessibilityLabel="CheckBox"
                    accessible={true}
                    focusable={true}
                    onClick={[Function]}
                    onResponderGrant={[Function]}
                    onResponderMove={[Function]}
                    onResponderRelease={[Function]}
                    onResponderTerminate={[Function]}
                    onResponderTerminationRequest={[Function]}
                    onStartShouldSetResponder={[Function]}
                    style={
                      Object {
                        "alignItems": "center",
                        "backgroundColor": "#FFFFFF",
                        "borderColor": "#FF0000",
                        "borderRadius": 4,
                        "borderWidth": 1,
                        "height": 24,
                        "justifyContent": "center",
                        "marginTop": 4,
                        "opacity": 1,
                        "width": 24,
                      }
                    }
                    testID="CheckBox"
                  >
                    <Image
                      resizeMode="contain"
                      source={
                        Object {
                          "ReactComponent": [Function],
                        }
                      }
                      style={
                        Object {
                          "height": 22,
                          "width": 22,
                        }
                      }
                    />
                  </View>
                </View>
                <Text
                  style={
                    Object {
                      "color": "#FF0000",
                      "fontFamily": "SofiaProRegular",
                      "fontSize": 12,
                      "lineHeight": 20,
                      "marginLeft": 16,
                    }
                  }
                >
                  [missing "en.CONTACT_DETAILS_SCREEN_STRINGS.AGREE_LABEL" translation]
                   
                </Text>
                <View
                  accessible={true}
                  focusable={true}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "marginLeft": -13,
                      "opacity": 1,
                    }
                  }
                >
                  <Text
                    style={
                      Array [
                        Object {
                          "color": "#FF0000",
                          "fontFamily": "SofiaProRegular",
                          "fontSize": 12,
                          "lineHeight": 20,
                          "marginLeft": 16,
                        },
                        Object {
                          "fontFamily": "SofiaPro-Bold",
                        },
                      ]
                    }
                  >
                    [missing "en.CONTACT_DETAILS_SCREEN_STRINGS.PRIVACY_POLICY_LABEL" translation]
                     
                  </Text>
                </View>
              </View>
              <View
                accessible={true}
                focusable={true}
                onClick={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Object {
                    "flexDirection": "row",
                    "marginLeft": 40,
                    "opacity": 1,
                  }
                }
              >
                <Text>
                  & 
                </Text>
                <Text
                  style={
                    Array [
                      Object {
                        "color": "#FF0000",
                        "fontFamily": "SofiaProRegular",
                        "fontSize": 12,
                        "lineHeight": 20,
                        "marginLeft": 16,
                      },
                      Object {
                        "fontFamily": "SofiaPro-Bold",
                        "marginLeft": 0,
                      },
                    ]
                  }
                >
                  [missing "en.CONTACT_DETAILS_SCREEN_STRINGS.T_AND_C_LABEL" translation]
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </RCTScrollView>
  </View>
  <View
    style={
      Object {
        "height": 118,
        "justifyContent": "space-between",
        "width": 722,
      }
    }
  >
    <View
      style={
        Array [
          Object {
            "alignItems": "center",
            "borderRadius": 4,
            "height": 52,
            "justifyContent": "center",
            "shadowColor": "#DDDDDD",
            "shadowOffset": Object {
              "height": 8,
              "width": 0,
            },
            "shadowOpacity": 0.75,
            "shadowRadius": 4,
          },
          Object {
            "backgroundColor": "#565655",
          },
        ]
      }
    >
      <View
        accessibilityLabel="Continue"
        accessible={true}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "#565655",
            "borderRadius": 4,
            "height": "100%",
            "justifyContent": "center",
            "opacity": 1,
            "width": 718,
          }
        }
        testID="Continue"
      >
        <Text
          numberOfLines={1}
          style={
            Array [
              Object {
                "fontFamily": "SofiaProRegular",
                "fontSize": 17,
                "textAlign": "center",
              },
              Object {
                "color": "#FFFFFF",
              },
            ]
          }
        >
          [missing "en.LOGIN_SCREEN_STRINGS.CONTINUE_LABEL" translation]
        </Text>
      </View>
    </View>
  </View>
</View>
`;
